week_2/dagster_ucr/project/week_2.py

```diff
@@ -5,26 +5,40 @@
 from dagster_ucr.resources import mock_s3_resource, redis_resource, s3_resource
 
 
-@op
-def get_s3_data():
-    pass
-
+@op(
+    config_schema={"s3_key": str},
+    required_resource_keys={"s3"},
+    out={"stocks": Out(dagster_type=List[Stock], description="List of Stocks")},
+)
+def get_s3_data(context):
+    output = list()
+    data = context.resources.s3.get_data(context.op_config['s3_key'])
+    for row in data:
+        output.append(Stock.from_list(row))
+    return output
 
-@op
-def process_data():
-    # Use your op from week 1
-    pass
+@op(ins={"stocks": In(dagster_type=List[Stock])},
+    out={'aggregation': Out(dagster_type=Aggregation)},
+    description="Given a list of stocks return the Aggregated values")
+def process_data(stocks):
+    highest = max([stock.high for stock in stocks])
```

Comment:
You could have gotten the object Stock and from it get the date and value, instead of doing 2 for loops, but since this is only a course, and it works...